string mode
string[] types
geometry_msgs/Vector3[] vals